python/podio_schema_evolution.py

```diff
@@ -0,0 +1,261 @@
+                                 (change.name, change.member_name,
+                                  change.old_member.full_type, change.new_member.full_type))
+
+        # are there dropped/added datatype pairs that could be interpreted as rename?
+        # for now assuming no change to the individual datatype definition
+        # I do not think more complicated heuristics are needed at this point in time
+        dropped_datatypes = [change for change in schema_changes if type(change) == DroppedDatatype]
+        added_datatypes = [change for change in schema_changes if type(change) == AddedDatatype]
+
+        for dropped in dropped_datatypes:
+            dropped_members = {member.name: member for member in dropped.datatype["Members"]}
+            for added in added_datatypes:
+                added_members = {member.name: member for member in added.datatype["Members"]}
+                if set(dropped_members.keys()) == set(added_members.keys()):
+                    self.warnings.append("Potential rename of '%s' into '%s'." % (dropped.name, added.name))
+
+        # are there dropped/added component pairs that could be interpreted as rename?
+        dropped_components = [change for change in schema_changes if type(change) == DroppedComponent]
+        added_components = [change for change in schema_changes if type(change) == AddedComponent]
+
+        for dropped in dropped_components:
+            dropped_members = {member.name: member for member in dropped.component["Members"]}
+            for added in added_components:
+                added_members = {member.name: member for member in added.component["Members"]}
+                if set(dropped_members.keys()) == set(added_members.keys()):
+                    self.warnings.append("Potential rename of '%s' into '%s'." % (dropped.name, added.name))
+
+    def print(self):
+        print("Comparing datamodel versions %s and %s"
+              % (self.datamodel_new.schema_version,
+                 self.datamodel_old.schema_version)
+              )
+        print("Found %i schema changes:" % len(self.schema_changes))
+        for change in self.schema_changes:
+            print(" - %s" % change)
+
+        if len(self.warnings) > 0:
+            print("Warnings:")
+            for warning in self.warnings:
+                print(" - %s" % warning)
+
+        if len(self.errors) > 0:
+            print("ERRORS:")
+            for error in self.errors:
+                print(" - %s" % error)
+
+    def read(self) -> None:
+        self.datamodel_new = self.reader.read(self.yamlfile_new, package_name="new")
+        self.datamodel_old = self.reader.read(self.yamlfile_old, package_name="old")
+        self.components1 = self.datamodel_new.components
+        self.components2 = self.datamodel_old.components
+        self.datatypes1 = self.datamodel_new.datatypes
+        self.datatypes2 = self.datamodel_old.datatypes
```

Comment:
Is storing these in member variables necessary? They are only used in one place, where it would be rather easy (and more legible) to just use `self.datamodel_new.components` or similar.